lib/Service/SettingsService.php

```diff
@@ -47,6 +47,16 @@ public function __construct(
 				},
 			],
 			'noteMode' => $this->getListAttrs('edit', 'preview'),
+			'customSuffix' => [
+				'default' => "",
+				'validate' => function ($value) {
+					$out = "";
+					if (preg_match('/^\.[A-Za-z0-9.-]*$/', $value)) {
+						$out = $value;
+					}
+					return $out;
+				},
```

Comment:
I did some tests with unexpected inputs. In this case, it results in an empty extension, which is inconvenient. What about the following solution?
1. it does not block wrong inputs, but removes illegal characters only.
2. if the result is empty, it uses `.txt` as default.

```suggestion
				'default' => '.txt',
				'validate' => function ($value) {
					$out = ltrim(preg_replace('/[^A-Za-z0-9.-]/', '', $value), '.');
					if (empty($out)) {
						$out = 'txt';
					}
					return '.' . $out;
				},
```

Reply:
Looks fine to me ! I didn't realize that a "wrong" extension would result in an empty one. Nice catch !